myapp/app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,17 @@
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def log_out
+    session.delete(:user_id)
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
```

Comment:
```suggestion
    @current_user ||= User.find(session[:user_id])
```
にしたいですね。
find_byだとnilを返してしまうので`find_by!`にして`RecordNotFound`エラー起こした方がよくて、
idで照会するので`find`を使うことにするといいです。

http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by-21
http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find

Reply:
`@current_user`はheaderなどでも使っており、
非ログイン状態でも参照されていて、RecordNotFoundが出てしまうと、処理が止まってしまいました。。
上記を変更したとしても、別途同じような処理は作らなければいけないため、
一旦上記を維持させていただけると助かります:pensive:

Reply:
> @current_userはheaderなどでも使っており、
> 非ログイン状態でも参照されていて、RecordNotFoundが出てしまうと、処理が止まってしまいました。。

なるほど。であればこれでいいと思います。